oak_functions/location_utils/src/tests.rs

```diff
@@ -16,165 +16,112 @@
-    let equator = find_cell(0.7, -120.3).unwrap().index;
-    assert_eq!(equator.row, 0);
-    assert_eq!(equator.col, 239);
-
-    // Near 45 degreees.
-    let halfway = find_cell(-45.2, 142.5).unwrap().index;
-    assert_eq!(halfway.row, -46);
-    assert_eq!(halfway.col, 100);
-
-    // Polar circle.
-    let polar = find_cell(-89.2, 50.1).unwrap().index;
-    assert_eq!(polar.row, -90);
-    assert_eq!(polar.col, 0);
-
-    // Invalid values.
-    let lat_too_small = find_cell(-90.1, 88.7);
-    assert!(lat_too_small.is_err());
-    let lat_too_big = find_cell(90.1, 88.7);
-    assert!(lat_too_big.is_err());
-    let lon_too_small = find_cell(-80.1, -180.7);
-    assert!(lon_too_small.is_err());
-    let lon_too_big = find_cell(80.1, 180.7);
-    assert!(lon_too_big.is_err());
-}
+fn test_location_from_degrees() {
+    let location = location_from_degrees(51.5074_f64, -0.1278_f64);
 
-#[test]
-fn test_final_cell_width() {
-    for lat in -90..90 {
-        let first_cell = find_cell(lat as f32, 0.0).unwrap();
-        let last_cell = find_cell(lat as f32, -1e-9).unwrap();
-        // Ensure both cells agree on the number of columns.
-        assert_eq!(first_cell.col_count, last_cell.col_count);
-        // Ensure that the last cell was really found.
-        assert_eq!(last_cell.index.col, last_cell.col_count - 1);
-        let actual_last_cell_width = 360.0 - (first_cell.width * last_cell.index.col as f32);
-        // Make sure that the difference between the width of the last cell and widths of the others
-        // is within 1%.
-        assert!((first_cell.width - actual_last_cell_width).abs() / first_cell.width < 0.01);
-    }
+    assert_eq!(Into::<E6>::into(&location.lat), E6(51_507_400));
+    assert_eq!(Into::<E6>::into(&location.lng), E6(-127_800));
 }
 
 #[test]
-fn test_index_key_round_trip() {
-    let initial = IndexKey { row: 30, col: -10 };
-    let result = IndexKey::from_bytes(&initial.to_bytes());
-    assert_eq!(initial, result);
+fn test_location_from_bytes() {
+    // Valid bytes.
+    let location = location_from_bytes(&[3, 17, 240, 200, 255, 254, 12, 200]).unwrap();
+
+    assert_eq!(Into::<E6>::into(&location.lat), E6(51_507_400));
+    assert_eq!(Into::<E6>::into(&location.lng), E6(-127_800));
+
+    // Bytes too short.
+    let location_result = location_from_bytes(&[3, 17, 240, 200, 255, 254, 12]);
+
+    assert!(location_result.is_err());
+
+    // Bytes too long.
+    let location_result = location_from_bytes(&[3, 17, 240, 200, 255, 254, 12, 200, 0]);
+
+    assert!(location_result.is_err());
 }
 
 #[test]
-fn test_index_value_round_trip() {
-    let initial = IndexValue {
-        value_key: [8, 7, 6, 5, 4, 3, 2, 1],
-        position: Point { x: 99, y: -999 },
-    };
-    let result = IndexValue::from_bytes(&initial.to_bytes());
-    assert_eq!(initial, result);
+fn test_location_to_bytes() {
+    // London.
+    let bytes = location_to_bytes(&location_from_degrees(51.5074_f64, -0.1278_f64));
```

Comment:
nit: Could these coordinates be defined as constants, since they are used in a few places in the tests? It helps to know the coordinates are valid and not some random values.